seqr/views/apis/dataset_api_tests.py

```diff
@@ -190,12 +192,11 @@ def test_add_variants_dataset(self, mock_es_search, mock_get_index_metadata, moc
         self.assertSetEqual(set(response_json.keys()), {'samplesByGuid', 'individualsByGuid', 'familiesByGuid'})
         sv_sample_guid = 'S1234567_NA19675_1'
         self.assertDictEqual(response_json['familiesByGuid'], {})
-        self.assertListEqual(response_json['samplesByGuid'].keys(), [sv_sample_guid])
+        self.assertListEqual(list(response_json['samplesByGuid'].keys()), [sv_sample_guid])
         self.assertEqual(response_json['samplesByGuid'][sv_sample_guid]['datasetType'], 'SV')
         self.assertTrue(response_json['samplesByGuid'][sv_sample_guid]['isActive'])
-        self.assertDictEqual(response_json['individualsByGuid'], {
-            'I000001_na19675': {'sampleGuids': [sv_sample_guid, existing_index_sample_guid]},
-        })
+        self.assertIn(sv_sample_guid, response_json['individualsByGuid']['I000001_na19675']['sampleGuids'])
+        self.assertIn(existing_index_sample_guid, response_json['individualsByGuid']['I000001_na19675']['sampleGuids'])
```

Comment:
So this somewhat changes what is tested, because before it was testing a) that exactly one individual is returned, b) that the only key returned for the individual is `sampleGuids` and c) that there are exactly 2 sample guids. This updated test (which I assume is because the order of those guids isn't stable) would pass if something changed and for whatever reason unwanted additional samples or individuals were returned or additional individual fields were included, but we want it to fail if that changes. What I would recommend to continue to have the same level of testing without relying on a list order is:
```
self.assertListEqual(list(individualsByGuid['individualsByGuid'].keys()), ['I000001_na19675'])
self.assertListEqual(list(individualsByGuid['individualsByGuid']['I000001_na19675'].keys()), ['sampleGuids'])
self.assertSetEqual(set(response_json['individualsByGuid']['I000001_na19675']['sampleGuids']), {sv_sample_guid, existing_index_sample_guid})
```

Reply:
The change was because of the list order issue. Agree with your points. Update as your recommendation.